pyomo/contrib/mpc/data/series_data.py

```diff
@@ -152,6 +155,44 @@ def get_data_at_time(self, time=None, tolerance=0.0):
             indices = indices[0]
         return self.get_data_at_time_indices(indices)
 
+    def get_interpolated_data_at_time(self, time=None):
+        """
+        Returns the data associated with the provided time point or points by
+        linear interpolation.
+
+        Parameters
+        ----------
+        time: Float or iterable
+            The time point or points corresponding to returned data.
+
+        Returns
+        -------
+        TimeSeriesData or ~scalar_data.ScalarData
+            TimeSeriesData containing only the specified time points
+            or dict mapping CUIDs to values at the specified scalar time
+            point.
+
+        """
+        if time is None:
+            # If time is not specified, assume we want the entire time
+            # set. Skip all the overhead, don't create a new object, and
+            # return self.
+            return self
+        is_iterable = _is_iterable(time)
+        if not is_iterable:
+            time = [time]
+        idxs = _get_time_index_vec(time, self._time)
```

Comment:
What happens if `time` contains points outside our `TimeSeriesData`'s range of time points? I think I'd want an error to be raised in this case. Can you add tests for cases with time points lower and higher than the acceptable range?